### ISSUE TITLE
When in debug REPL, tab key changes focus instead of indents

### ISSUE THREAD

**Issue by mhkeller** [microsoft/vscode#169798]
Does this issue occur when all extensions are disabled?: Yes/No

ersion: 1.74.2
Commit: e8a3071ea4344d9d48ef8a4df2c097372b0c5161
Date: 2022-12-20T10:26:09.430Z (1 day ago)
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 21.6.0
Sandboxed: No

Steps to Reproduce:

1. Put the cursor into the Debug REPL
2. Press the tab key. The focus will change to the bottom bar instead of adding an indentation

Is there any way to change this behavior so that the tab key adds an indent?

**Comment by roblourens** [comment]
In chrome devtools, tab will tab out of the input when the cursor is on the first line, and insert a tab when it's on the 2nd or below line. Do you think this is sufficient for accessibility requirements @isidorn? 

Or, should we have a mode like the editor that you have to toggle manually with ctrl+m?

**Comment by isidorn** [comment]
@roblourens yes, I think that is a reasonable experience. An alternative idea -> only when the cursor is at the very end of the debug input content the tab would focus out.
But whatever we do I would love if we do it for all SimpleEditorWidgets so we have a consistent behaviour across all our input boxes.

fyi @meganrogge

**Comment by mhkeller** [comment]
Here is another surprising behaviors that I'll add here since it's related (but happy to make new issues if that is easier to track)

* If you hit the up arrow to get your last statement, you can then hit command+down-arrow to get to the end of a multi-line statement. I thought that this would now put me into normal cursor editing mode where I could use up, down, left and right to navigate my command but that's not the case. If it hit up-arrow, the text in my REPL gets replaced with the previous command. 

In Chrome, the behavior is different. If I hit up arrow, it places my cursor at the end of the inserted statement and I can use my up, down, left and right keys as normal unless I key all the way up to the top line and hit the up arrow. That UX feels more intuitive to me since my expectation is that if my cursor is, say, on line 3 of 5 and I hit the up arrow, the cursor will go to line 2 and not replace my statement with something else.

**Comment by mhkeller** [comment]
In short, if the REPL could have the same UX as chrome dev tools that would be helpful for muscle memory and for navigating these expectations of how the interface will react.

**Comment by mhkeller** [comment]
Hi, Just wanted to check in to see if it would be possible to have this added to a release target? I run into this issue frequently.

**Comment by mhkeller** [comment]
Just checking in to note that this would still be a useful feature. Another area where it doesn't match Chrome is the REPL area is very short on the bottom. It would be easier to use if you had a larger text area to type in, like how Chrome lets you immediately after your last returned statement instead of in a bar at the bottom. Any time you can spend on this would be much appreciated.
